x/concentrated-liquidity/genesis.go

```diff
@@ -79,6 +79,16 @@ func (k Keeper) InitGenesis(ctx sdk.Context, genState genesis.GenesisState) {
 		feePositionKey := types.KeyFeePositionAccumulator(positionWrapper.Position.PositionId)
 
 		k.initOrUpdateAccumPosition(ctx, feeAccumObject, positionWrapper.FeeAccumRecord.AccumValuePerShare, feePositionKey, positionWrapper.FeeAccumRecord.NumShares, positionWrapper.FeeAccumRecord.UnclaimedRewardsTotal, positionWrapper.FeeAccumRecord.Options)
+
+		positionName := string(types.KeyPositionId(positionWrapper.Position.PositionId))
+		uptimeAccumulators, err := k.GetUptimeAccumulators(ctx, positionWrapper.Position.PoolId)
+		if err != nil {
+			panic(err)
+		}
+
+		for uptimeIndex, uptimeRecord := range positionWrapper.UptimeAccumRecords {
+			k.initOrUpdateAccumPosition(ctx, uptimeAccumulators[uptimeIndex], uptimeRecord.AccumValuePerShare, positionName, uptimeRecord.NumShares, uptimeRecord.UnclaimedRewardsTotal, uptimeRecord.Options)
```

Comment:
Is there a reason we use this function instead of `initOrUpdatePositionUptimeAccumulators` (and `initOrUpdatePositionFeeAccumulator` for fees)?

Not sure what the ultimate use of these records are so it might not matter, but this abstraction sets the records directly in state and thus circumvents important internal variable updates (e.g. calling `GetTotalShares` on the accums would return 0 even after the records were created)

Reply:
the way i thought of this was this; 
When we `initGenesis` at the very beginning all the values are set to 0. If we `exportGenesis` and then run `initGenesis` we get the latest data that includes "internal variable updates" and we just set new genesis to be the exported value

thus no need to run complex init logic since if export handles it by getting latest values

Reply:
I see, thanks for the clarification